### PR TITLE
realbrk.cpp : Fix sprite priority, Add notes, Reduce runtime tag lookups

### PR DESCRIPTION
**Pull request by cam900**

### CODE REVIEW

File: src/mame/video/realbrk.cpp [hunk 1]
```diff
@@ -236,6 +213,8 @@ void realbrk_state::draw_sprites(bitmap_ind16 &bitmap,const rectangle &cliprect)
 		attr    =       s[ 6 ];
 		code    =       s[ 7 ];
 
+		if(( flip & 0x03 ) != layer ) continue;
```
Comment (angelosa) [inline src/mame/video/realbrk.cpp]: If it controls priority then it's not flip :)

**Comment by angelosa** [comment]
Also there's something dumb in this driver, which you might eventually checkout:

`rot = flip & 0x30;
`
then it checks:
`(rot & 1 ? scaley : scalex) << 12, (rot & 1 ? scalex : scaley) << 12,0);`

Which of course can't be right, prolly some debugging that got lost in time?
Regardless, imho you should comment out the ternary operators, make a shorter version of it then call "flip" to something more descriptive of what it does (like attr or flags or w/e), since it's really obvious it doesn't control that only.

File: src/mame/video/realbrk.cpp [hunk 2]
```diff
@@ -213,14 +214,17 @@ void realbrk_state::draw_sprites(bitmap_ind16 &bitmap,const rectangle &cliprect,
...
+		if(pri != layer)
+			continue;
 
 		xnum    =       ((dim >> 0) & 0x1f) + 1;
 		ynum    =       ((dim >> 8) & 0x1f) + 1;
 
 		flipx   =       flip & 0x0100;
 		flipy   =       flip & 0x0200;
-		rot     =       flip & 0x0030;
+		rot     =       (flip & 0x0030) >> 4;
```
Comment (angelosa) [inline src/mame/video/realbrk.cpp]: This one now actively changes implementation as stated before.
Are you sure it's like this?

Comment (cuavas) [reply src/mame/video/realbrk.cpp]: Check the code below - he adjusted the switch statement for the shift.

Comment (angelosa) [reply src/mame/video/realbrk.cpp]: Overlooked it, github sucks lol

**Pull request merged**